tasks/data_handling/tests/test_data_pre_processing.py

```diff
@@ -0,0 +1,283 @@
+import scanpy as sc
+import unittest
+import os
+import numpy as np
+from collections import Counter
+
+import global_settings as gs
+from tasks.data_handling import data_pre_processing
+from tasks.test_helper.builders.data_handling import data_pre_process as test_data_pre_process
+from tasks.test_helper.builders.utils import context_utils as cu
+
+
+class TestCreateCountMatrix(unittest.TestCase):
+    def test_need_file_path(self):
+        with self.assertRaises(ValueError) as context:
+            result = data_pre_processing.create_count_matrix()
+
+    data_path = os.path.join('\\'.join(os.getcwd().split('\\')[:-2]), gs.TEST_DATA_PATH)
+
+    def tearDown(self) -> None:
+        cu.clean(folder_path=self.data_path)
+
+    file_name = 'simulated_csc_h5_data.h5'
+    simulated_matrix = test_data_pre_process.simulate_csc_h5_data(file_path=data_path,
+                                                                  filename=file_name)
+    simulated_adata = data_pre_processing.create_count_matrix(file_path=data_path + '\\' + file_name)
+
+    def test_equal_values_equal_index(self):
+        equal = True
+        indices = self.simulated_adata.X.nonzero()
+        rows = indices[0]
+        columns = indices[1]
+
+        for i in range(len(rows)):
+            row = rows[i]
+            col = columns[i]
+            if self.simulated_adata.X[row, col] != self.simulated_matrix.A[row, col]:
```

Comment:
I don't understand why are you comparing two simulated matrices to each other

Reply:
I compare it to make sure that the `data_pre_processing.create_count_matrix()` do as expected, i.e. doesn't do anything funky with the indices or the values

Reply:
If that's the case, I have two questions:

1. How's this different from what we already have? (See image)
![image](https://github.com/lozadaa-chalmers/transformer-bio/assets/144219103/0210830b-03fd-41bb-93be-f645b3d6469d)

2. I understand why it makes sense to make the comparison now. Just don't do it element by element. Compare both matrices directly (see picture on how to do this).

![image](https://github.com/lozadaa-chalmers/transformer-bio/assets/144219103/213b56e7-47d7-49e4-a296-02b0cbd0442d)

Reply:
The difference is that it creates a count_matrix using scanpy which also gives it some extra functions, e.g. get the gene names and such such that it is compatible with the rest of the pre-processing steps.